src/model_getters.jl

```diff
@@ -359,3 +359,59 @@ function ThermoDispatcher(atmos)
         compressibility_model,
     )
 end
+
+function AbstractCovarianceModel(paramset::NamedTuple)
+    thermo_covariance_model_name = paramset.thermo_covariance_model
+    if thermo_covariance_model_name == "prognostic"
+        return PrognosticThermoCovariances()
+    elseif thermo_covariance_model_name == "diagnostic"
+        covar_lim = paramset.diagnostic_covar_limiter
+        return DiagnosticThermoCovariances(covar_lim)
+    else
+        error(
+            "Something went wrong. Invalid thermo_covariance model: '$thermo_covariance_model_name'",
+        )
+    end
+end
+
+function AbstractPrecipFractionModel(paramset::NamedTuple)
+    precip_fraction_model_name = paramset.precip_fraction_model
+    if precip_fraction_model_name == "prescribed"
+        return PrescribedPrecipFraction(paramset.prescribed_precip_frac)
+    elseif precip_fraction_model_name == "cloud_cover"
+        return DiagnosticPrecipFraction(paramset.precip_fraction_limiter)
+    else
+        error(
+            "Something went wrong. Invalid `precip_fraction` model: `$precip_fraction_model_name`",
+        )
+    end
+end
+
+function AbstractQuadratureType(s::String)
```

Comment:
This is a confusing name, because Julia prohibits creating instances of abstract types, can we just call it `QuadratureType`?

Reply:
Also, why not just pass in a Dict and read it from the Dict? Can we also error if none of the cases are satisfied?

Reply:
Maybe it makes sense to just move it all into the TurbulenceConvectionParameters? It shouldn't get used anywhere else.